backends/kafka/lag.go

```diff
@@ -0,0 +1,160 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/batchcorp/plumber/cli"
+	"github.com/batchcorp/plumber/printer"
+	"github.com/pkg/errors"
+	skafka "github.com/segmentio/kafka-go"
+)
+
+// validate cli options and init connection
+func Lag(opts *cli.Options) error {
+	if err := validateLagOptions(opts); err != nil {
+		return errors.Wrap(err, "unable to validate read options")
+	}
+
+	kafkaLag, err := NewKafkaLagConnection(opts)
+
+	for _, v := range kafkaLag.partitionDiscoverConn {
+		defer v.Close()
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "unable to create connection")
+	}
+
+	groupId := opts.Kafka.GroupID
+
+	return kafkaLag.LagCalculationForConsumerGroup(groupId, opts)
+}
+
+// calculate lag with a given connection
+func (kLag *KafkaLag) LagCalculationForConsumerGroup(groupId string, opts *cli.Options) error {
+
+	topicPartionMap := make(map[string][]skafka.Partition)
+
+	for tp, kc := range kLag.partitionDiscoverConn {
+
+		partitionList, err := kc.ReadPartitions(tp)
+
+		if err != nil {
+			return errors.Wrap(err, "unable to obtain partitions")
+		}
+
+		topicPartionMap[tp] = partitionList
+	}
+
+	// calculate and print lag
+	var sb strings.Builder
+
+	for tpKey, partValue := range topicPartionMap {
+
+		for _, part := range partValue {
+
+			lagPerPartition, err := kLag.LagCalculationPerPartition(tpKey, groupId, part.ID, opts)

```

Comment:
Generally, you don't want an empty line between a method call and it's error check

Reply:
resolving in the next commit